tools/python_api/src_cpp/py_connection.cpp

```diff
@@ -217,14 +339,55 @@ Value transformPythonValue(py::handle val) {
         Interval::addition(interval, seconds, "seconds");
         Interval::addition(interval, microseconds, "microseconds");
         return Value::createValue<interval_t>(interval);
-    } else if (py::isinstance(val, uuid)) {
-        auto strVal = py::str(val).cast<std::string>();
-        auto uuidVal = UUID::fromString(strVal);
-        ku_uuid_t uuidToAppend;
-        uuidToAppend.value = uuidVal;
-        return Value{uuidToAppend};
-    } else {
-        throw std::runtime_error(
-            "Unknown parameter type " + py::str(val.get_type()).cast<std::string>());
     }
+    case LogicalTypeID::UUID: {
+        auto ptr = val.ptr();
+        auto days = PyDateTime_DELTA_GET_DAYS(ptr);
+        auto seconds = PyDateTime_DELTA_GET_SECONDS(ptr);
+        auto microseconds = PyDateTime_DELTA_GET_MICROSECONDS(ptr);
+        interval_t interval;
+        Interval::addition(interval, days, "days");
+        Interval::addition(interval, seconds, "seconds");
+        Interval::addition(interval, microseconds, "microseconds");
+        return Value::createValue<interval_t>(interval);
+    }
+    case LogicalTypeID::VAR_LIST: {
+        py::list lst = py::reinterpret_borrow<py::list>(val);
+        std::vector<std::unique_ptr<Value>> children;
+        for (auto child : lst) {
+            children.push_back(std::make_unique<Value>(
+                transformPythonValueAs(child, VarListType::getChildType(type))));
+        }
+        return Value(std::make_unique<LogicalType>(*type), std::move(children));
+    }
+    case LogicalTypeID::MAP: {
+        py::dict dict = py::reinterpret_borrow<py::dict>(val);
```

Comment:
Do you have to acquire a lock there?

Reply:
Since we're borrowing the object, [no change is made to the refcount](https://docs.python.org/2/c-api/intro.html#reference-count-details), so the GIL shouldn't be needed